rust/src/web/graphql/next_stakes/mod.rs

```diff
@@ -0,0 +1,241 @@
+                        return *public_key == account.pk.0;
+                    }
+                    if let Some(delegate) = &query.delegate {
+                        return *delegate == account.delegate.0;
+                    }
+                    if let Some(query_ledger_hash) = query_ledger_hash {
+                        return *query_ledger_hash == ledger_hash;
+                    }
+                    if let Some(query_epoch) = query_epoch {
+                        return *query_epoch == epoch;
+                    }
+                    if let Some(delegate) = delegate {
+                        return *delegate == account.delegate.0;
+                    }
+                }
+                true
+            })
+            .map(|account| {
+                let pk = account.pk.clone();
+                let result = delegations.delegations.get(&pk).unwrap();
+                let total_delegated_nanomina = result.total_delegated.unwrap_or_default();
+                let count_delegates = result.count_delegates.unwrap_or_default();
+                let mut decimal = Decimal::from(total_delegated_nanomina);
+                decimal.set_scale(9).ok();
+
+                let total_delegated = decimal.to_f64().unwrap_or_default();
+
+                let timing = account.timing.as_ref().map(|timing| NextStakesTiming {
+                    cliff_amount: Some(timing.cliff_amount),
+                    cliff_time: Some(timing.cliff_time),
+                    initial_minimum_balance: Some(timing.initial_minimum_balance),
+                    vesting_increment: Some(timing.vesting_increment),
+                    vesting_period: Some(timing.vesting_period),
+                });
+
+                NextStakesLedgerAccountWithMeta {
+                    epoch,
+                    ledger_hash: ledger_hash.clone(),
+                    account: NextStakesLedgerAccount::from(account),
+                    delegation_totals: NextStakesDelegationTotals {
+                        total_delegated,
+                        total_delegated_nanomina,
+                        count_delegates,
+                    },
+                    timing,
+                }
+            })
+            .collect();
+
+        match sort_by {
+            Some(NextStakesSortByInput::BalanceAsc) => {
+                accounts.sort_by(|b, a| b.account.balance_nanomina.cmp(&a.account.balance_nanomina))
+            }
+            Some(NextStakesSortByInput::BalanceDesc) => {
+                accounts.sort_by(|a, b| b.account.balance_nanomina.cmp(&a.account.balance_nanomina))
+            }
+            None => (),
+        }
+
+        accounts.truncate(limit);
+        Ok(Some(accounts))
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct NextStakesLedgerAccountWithMeta {
+    /// Value next epoch
+    epoch: u32,
+    /// Value next ledger hash
+    ledger_hash: String,
+    /// Value delegation totals
+    delegation_totals: NextStakesDelegationTotals,
```

Comment:
```suggestion
    #[graphql(name = "nextDelegationTotals")]
    next_delegation_totals: NextStakesDelegationTotals,
```

Reply:
The explicit naming shouldn't be necessary since snake_case gets automatically converted to camelCase